### PR TITLE
feat(cost-optimization-report): add openshift cost optimization report

### PR DESCRIPTION
**Pull request by hemslo**
Export openshift cost optimization report from [Cost Management Optimizations](https://console.redhat.com/openshift/cost-management/optimizations).

Preview a sample report at [test cost optimization report](https://gitlab.cee.redhat.com/-/snippets/8885).

Local verification:

```shell
qontract-cli --config config.toml get openshift-cost-optimization-report > openshift-cost-optimization-report.md
```

[design doc](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-sre/design-docs/openshift-cost-report.md#for-cost-optimization)

[APPSRE-10233](https://issues.redhat.com/browse/APPSRE-10233)

### CODE REVIEW

**Review by rporres — APPROVED** [review]
This looks great. The commit by commit review makes this review a great experience, thx a lot.

Is there a possibility of avoiding horizontal scrolling in the rendered report? Or is it something up to gitlab. It makes reading some tables a bit complicated.

File: tools/cli_commands/test/fixtures/openshift_cost_optimization_report.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+Report is generated for optimization enabled namespaces
+(`insights_cost_management_optimizations='true'` in namespace `labels`).
+
+In Cost optimizations, recommendations get generated when
+CPU usage is at or above the 60th percentile and
+memory usage is at the 100th percentile.
+
+View details in [Cost Management Optimizations](https://console.redhat.com/openshift/cost-management/optimizations).
+
+## app
```
[the remark this answers is not in the document]
Comment (hemslo) [reply tools/cli_commands/test/fixtures/openshift_cost_optimization_report.md:14]: this is the name of an `app` (`service`), in test data, the value is set [here](https://github.com/app-sre/qontract-reconcile/pull/4734/files#diff-10d4937944318a95d83d7003ae0db9c8f0c6e45346c1ce3d24711e662b7cb4d1R100)

**Comment by hemslo** [comment]
> Is there a possibility of avoiding horizontal scrolling in the rendered report? Or is it something up to gitlab. It makes reading some tables a bit complicated.

@rporres the json table in gitlab is not easily controlled by us, as it's a gitlab feature, the plan is to export this data somewhere else (most likely prometheus), then setup alert and grafana dashboard for it.

**Comment by rporres** [comment]
> @rporres the json table in gitlab is not easily controlled by us, as it's a gitlab feature, the plan is to export this data somewhere else (most likely prometheus), then setup alert and grafana dashboard for it.

Yeah, it looked like it wasn't easy. And I like the plan, the data is very interesting but it will be easier to digest that way.

**Pull request merged**